Source/WebGPU/WebGPU/BindGroup.mm

```diff
@@ -116,6 +122,13 @@ static MTLRenderStages metalRenderStage(ShaderStage shaderStage)
                 id<MTLTexture> texture = WebGPU::fromAPI(entry.textureView).texture();
                 [argumentEncoder[stage] setTexture:texture atIndex:index++];
                 resources.append({ texture, MTLResourceUsageRead, metalRenderStage(stage) });
+            } else if (externalTextureIsPresent) {
+                auto pixelBuffer = WebGPU::fromAPI(externalTexture).pixelBuffer();
+                id<MTLTexture> texture = createTextureFromPixelBuffer(pixelBuffer);
+                [argumentEncoder[stage] setTexture:texture atIndex:index++];
```

Comment:
I don't understand. The whole point of external textures is that they are "fat" objects, that include way more than just a texture. I expected to see a shading language component, where sample operations on external textures get codegen'ed to a big block of code.

Even if that isn't all implemented in this first patch, I think we should still think about where/how to handle "fat" textures, and leave a space here to be filled-in with the relevant code that will come in a follow-up patch.

Reply:
By using the MTLPixelFormat SPI we avoid need for this code-gen, it makes it simpler and more performant